app/qml/form/FeatureToolbar.qml

```diff
@@ -27,16 +27,19 @@ Item {
             name: "edit"

```

Comment:
Hi @alexbruy, can I ask you to write a comment to explain each of these states? thanks! Otherwise it looks good

Reply:
Comments added for "edit" and "add" states. I struggle to figure out when "readOnly" is used.